### ISSUE TITLE
High CPU usage

### ISSUE THREAD

**Issue by nekinie** [swaywm/sway#1940]
I installed Arch Linux and Sway on my desktop and made a few tweaks to the configuration, all was well. After getting home from work today, If I start Sway, it consumes 100% CPU resources on all 16 threads. I disabled my custom configuration and the issue persists. I enabled verbose and debug logging but nothing obvious is being shown. I have a similar configuration running on my workstation without any troubles.

Sways version is 0.15.2.

If I find some time tonight, ill learn how to run some traces and see if I can find anything useful.

**Comment by SirCmpwn** [comment]
Try the master branch.

**Comment by ascent12** [comment]
This is a known issue with 0.15. Sway 1.0-alpha has done a lot to make the CPU usage significantly lower.

**Comment by nekinie** [comment]
I installed sway-git from the AUR (r2587.22db6013-1) and the issue is no longer present. There is noticeable screen tearing, specifically when moving the mouse but suspect that's been picked up in another ticket.

Thanks for the help!

**Comment by petoknm** [comment]
I remember having the same issue once when my nvidia graphics card complained about something called "falcon" during boot... I remember i had to power off the machine wait a bit and power on... Fortunately it happened to me only once...

**Comment by nekinie** [comment]
I have Nvidia cards on the desktop using the open source drivers if it helps, the workstation is running AMD cards.

**Comment by SirCmpwn** [comment]
The sway-git package is on the 0.15 branch. Build sway yousrelf.

**Comment by RedBorg** [comment]
I have the same problem, running sway built from master. I am running Parabola ( *libre*-patched arch linux ) with *nouveau* drivers, if it helps. Sway will use near to 98% on all cores.

**Comment by ddevault** [comment]
What version of sway?

**Comment by RedBorg** [comment]
> What version of sway?

Built from master branch yesterday, wlroots from *wlroots-git* package in aur.

**Comment by RedBorg** [comment]
Driver issue pretty much confirmed to me. Recreated setup on another computer and worked flawlessly

**Comment by sposterkil** [comment]
I'm seeing
